### PR TITLE
feat: Add generic test ComponentWrap

### LINKED ISSUE #1332: Create generic test wrapper for components
Implement a component test wrapper to be used for any `Component` containing methods for testing the component.
Wrapper should contain the component instance (which should be gettable) and common methods needed for testing.

The `isUsable` test method should be available that is used to check if the component should be interactable by the user
on the client side which requires that the component and parents are enabled, visible, attached and not inert.

Abstract UITest class should get a method for getting the wrapper for a given component 
for example `public <T extends ComponentWrap<Y>, Y extends Component> T $(Y component)`

Acceptance criteria
- Generic test wrapper class exists
- Documentation on usage

### PR DESCRIPTION
**Pull request by caalador**
Add a generic test wrapper for
Components with usability check.

Fixes #1332

### CODE REVIEW

File: vaadin-testbench-core/src/main/java/com/vaadin/testbench/unit/ComponentWrap.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ * <p>
+ * More targeted methods for specific components exist in named component
+ * wrappers.
+ *
+ * @param <T>
+ *         component type
+ */
+public class ComponentWrap<T extends Component> {
+
+    private T component;
```
Comment (mcollovati) [inline vaadin-testbench-core/src/main/java/com/vaadin/testbench/unit/ComponentWrap.java:28]: Can be final?

File: vaadin-testbench-core/src/test/java/com/vaadin/testbench/unit/ComponentWrapTest.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+                "Home should be visible and interactable");
+    }
+
+    @Test
+    public void componentIsDisabled_isUsableReturnsFalse() {
+        home.getElement().setEnabled(false);
+
+        final ComponentWrap<Home> home_ = $(home);
+        Assertions.assertFalse(home_.isUsable(),
+                "Home should be visible and interactable");
```
Comment (mcollovati) [inline vaadin-testbench-core/src/test/java/com/vaadin/testbench/unit/ComponentWrapTest.java:49]:
```suggestion
                "Home should be visible but not interactable");
```

File: vaadin-testbench-core/src/test/java/com/vaadin/testbench/unit/ComponentWrapTest.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    }
+
+    @Test
+    public void componentIsHidden_isUsableReturnsFalse() {
+        home.setVisible(false);
+
+        final ComponentWrap<Home> home_ = $(home);
+        Assertions.assertFalse(home_.isUsable(),
+                "Home should be visible and interactable");
```
Comment (mcollovati) [inline vaadin-testbench-core/src/test/java/com/vaadin/testbench/unit/ComponentWrapTest.java:58]:
```suggestion
                "Home should not be interactable when component is not visible");
```

File: vaadin-testbench-core/src/test/java/com/vaadin/testbench/unit/ComponentWrapTest.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        home.add(span);
+        final ComponentWrap<Span> span_ = $(span);
+
+        Assertions.assertTrue(span_.isUsable(),
+                "Span should be attached to the ui");
+
+        span_.setModal(true);
+
+        Assertions.assertTrue(span_.isUsable(),
+                "Span should interactable when it is modal");
```
Comment (mcollovati) [inline vaadin-testbench-core/src/test/java/com/vaadin/testbench/unit/ComponentWrapTest.java:99]:
```suggestion
                "Span should be interactable when it is modal");
```

**Review by mcollovati — CHANGES_REQUESTED** [review]

File: vaadin-testbench-core/src/main/java/com/vaadin/testbench/unit/ComponentWrap.java (hunk 1 above)
Comment (caalador) [reply vaadin-testbench-core/src/main/java/com/vaadin/testbench/unit/ComponentWrap.java]: done

**Review by mcollovati — APPROVED** [review]

**Pull request merged**